experimental/dds/tree2/src/test/class-tree/schemaFactory.examples.spec.ts

```diff
@@ -0,0 +1,116 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "node:assert";
+
+import { MockFluidDataStoreRuntime } from "@fluidframework/test-runtime-utils";
+import { SchemaFactory, ITree, TreeConfiguration, TreeView, Tree } from "../../class-tree";
+import { TreeFactory } from "../../treeFactory";
+
+// Since this no longer follows the builder pattern its a SchemaFactory instead of a SchemaBuilder.
+const schema = new SchemaFactory("com.example");
+
+/**
+ * An example schema based type.
+ * This doc comment will be visible in intellisense when using referring to the schema (as a runtime value) or the instance type.
+ */
+class Point extends schema.object("Point", {
+	x: schema.number,
+	y: schema.number,
+}) {}
+
+class Note extends schema.object("Note", {
+	/**
+	 * Doc comment on a schema based field. Intellisense should work when referencing the field.
+	 */
+	text: schema.string,
+	/**
+	 * Example optional field.
+	 * Works the same as before.
+	 */
+	location: schema.optional(Point),
+}) {
+	// Schema based classes can have methods and extra session local properties.
+	// We don't have to recommend this as a design pattern, but it does just work.
+	public isSelected: boolean = false;
+
+	// Methods also work, and can access the tree APIs as normal:
+	public moveToFront(): void {
+		const parent = Tree.parent(this);
+		// instanceof just works: nothing special to learn for either JS or TS narrowing of unions.
+		if (parent instanceof NodeList) {
+			const key = Tree.key(this);
+			assert(typeof key === "number");
+			parent.moveToEnd(key);
+		}
+	}
+}
+
+class NodeMap extends schema.map("NoteMap", Note) {}
+class NodeList extends schema.list("NoteList", Note) {}
+
+// Example function working on some schema based types.
+// Note that there is no need to use Typed<typeof NodeMap> anymore (either inline or as a separate type declaration).
+function f(n: NodeMap): Note[] {
+	// These schema based types can have methods provided based on their node kind. In this case maps have `get`.
+	// The type returned from `get` is `Note | undefined`, and shows that way in the intellisense:
+	// It does not show some complex type expression equivalent to that which is a big improvement over the previous setup.
+	const item: Note | undefined = n.get("x");
+	return item === undefined ? [] : [item];
+}
+
+class Canvas extends schema.object("Canvas", { stuff: [NodeMap, NodeList] }) {}
+
+// TODO: this should be possible, and this approach is needed in some cases where types are ambagious:
+const config = new TreeConfiguration(
+	Canvas,
+	() =>
+		new Canvas({
+			stuff: new NodeList([
+				{ text: "a", location: undefined },
+				new Note({ text: "b", location: undefined }),
+			]),
+		}),
+);
+// Currently the constructors of lists and maps cannot be used to make unhydrated nodes.
```

Comment:
Just to make sure I'm understanding correctly, `config` is what we want and `configWorkaround` is a temporary hack until we properly support `config`?

Reply:
Both should work, but the above version currently doesn't.